src/olympia/files/models.py

```diff
@@ -508,7 +508,7 @@ def add_file(self, chunks, filename, size):
         if ext in amo.VALID_ADDON_FILE_EXTENSIONS:
             ext = '.xpi'
         self.path = os.path.join(
-            user_media_path('addons'), 'temp', f'{uuid.uuid4().hex}{ext}'
+            settings.MEDIA_ROOT, 'fileuploads', f'{uuid.uuid4().hex}{ext}'
```

Comment:
```suggestion
            user_media_path('fileuploads'), f'{uuid.uuid4().hex}{ext}'
```

Reply:
mmm yes, but... I was kinda thinking about removing that utility function entirely. I find it more confusing that helpful...

Reply:
it _is_ a bit "magic"

Reply:
https://github.com/mozilla/addons-server/issues/18999